android/src/main/java/com/reactnativecommunityviewpager/ViewPagerViewManager.kt

```diff
@@ -0,0 +1,70 @@
+package com.reactnativecommunityviewpager
+
+import android.view.View
+import androidx.fragment.app.FragmentActivity
+import androidx.viewpager2.widget.ViewPager2
+import androidx.viewpager2.widget.ViewPager2.OnPageChangeCallback
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.ViewGroupManager
+import com.facebook.react.uimanager.annotations.ReactProp
+import com.facebook.react.uimanager.events.RCTEventEmitter
+
+class ViewPagerViewManager : ViewGroupManager<ViewPager2>() {
+  override fun getName() = "RNCViewPager"
+
+  override fun createViewInstance(reactContext: ThemedReactContext): ViewPager2 {
+    return ViewPager2(reactContext).apply {
+      adapter = ReactViewAdapter(reactContext.currentActivity as FragmentActivity)
+
+      registerOnPageChangeCallback(object : OnPageChangeCallback() {
+        override fun onPageSelected(position: Int) {
+          reactContext.getJSModule(RCTEventEmitter::class.java).receiveEvent(
+            id,
+            "topPageSelected",
```

Comment:
why `topPageSelected` ?

Reply:
If I understand correctly, that is what official docs state https://reactnative.dev/docs/native-components-android#events and how it is implemented on `next` https://github.com/callstack/react-native-viewpager/blob/next/android/src/main/java/com/reactnativecommunity/viewpager/event/PageSelectedEvent.java#L23 .